RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -1558,6 +1571,10 @@ boolean __restore(string resource_type, int goal, int meat_reserve, boolean useF
 
 	while(current_resource() < goal)
 	{
+		if(goal > max_resource())	//prevent infinite loop in case maxHP or maxMP dropped below goal
+		{
+			goal = max_resource();
+		}
```

Comment:
this will catch infininte loops not just when goal is over MP, but also when goal is over HP. which is also something that happened before